tests/test_models/test_harmonic_oscillators/test_initial_conditions.py

```diff
@@ -0,0 +1,35 @@
+import math
+
+import pytest
+
+from hamilflow.models.harmonic_oscillator.initial_conditions import parse_ic_for_sho
+
+
+@pytest.fixture()
+def omega() -> float:
+    return 2 * math.pi
+
+
+class TestParseICForSHO:
+    @pytest.mark.parametrize(
+        ("input", "expected"),
+        [
+            (
+                dict(x0=1.0, v0=1.0, phi=7.0),
+                dict(x0=1.0, v0=1.0, phi=7 % (2 * math.pi)),
+            ),
+            (dict(x0=1.0, t0=1.0), dict(x0=1.0, v0=0.0, phi=0.0)),
+            (
+                dict(E=1.0, t0=1.0),
+                dict(x0=math.sqrt(2.0) / (2 * math.pi), v0=0.0, phi=0.0),
+            ),
+        ],
+    )
+    def test_output(
+        self, omega: float, input: dict[str, float], expected: dict[str, float]
```

Comment:
Now I see what is not compatible with py39... It is `expected: dict[str, float]`. Should be `expected: Dict[str, float]`, where `Dict` is imported from `typing`.